### PR TITLE
ScrollView scroll to content edge freezing issue simulation.

### PR DESCRIPTION
**Pull request by Ad1991**
Changes to simulate the UIScrollView issue where scroll view gets stuck in infinite loop if you try to scroll to content edge and scrollView's content size is equal to or smaller than scrollView's own size. For more check this [conversation](https://groups.google.com/forum/#!topic/earlgrey-discuss/JaM4g9b5QFg).

### CODE REVIEW

**Comment by Ad1991** [comment]
I signed it!

File: Tests/FunctionalTests/Sources/FTRScrollViewTest.m [hunk 1]
```diff
@@ -28,7 +28,7 @@ - (void)setUp {
   [self openTestViewNamed:@"Scroll Views"];
 }
 
-- (void)testScrollToTopEdge {
+- (void)DISABLED_testScrollToTopEdge {
```
Comment (khandpur) [inline Tests/FunctionalTests/Sources/FTRScrollViewTest.m]: You should add another test so that it doesn't disable an existing test that's providing us with indication that scroll to top edge works for some scenarios.

File: Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib [hunk 2]
```diff
@@ -1,10 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="12121" systemVersion="16F73" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" colorMatched="NO">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="12121" systemVersion="16G29" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" colorMatched="YES">
+    <device id="retina4_7" orientation="portrait">
```
Comment (khandpur) [inline Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib]: remove device id stuff

File: Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
+        <adaptation id="fullscreen"/>
+    </device>
     <dependencies>
         <deployment identifier="iOS"/>
-        <development version="7000" identifier="xcode"/>
         <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="12089"/>
         <capability name="Constraints to layout margins" minToolsVersion="6.0"/>
+        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
```
Comment (khandpur) [inline Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib]: this means our CI that uses xcode 7 will break.

File: Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib [hunk 4]
```diff
@@ -19,66 +22,66 @@
         </placeholder>
         <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
         <view contentMode="scaleToFill" id="1">
-            <rect key="frame" x="0.0" y="0.0" width="600" height="600"/>
+            <rect key="frame" x="0.0" y="0.0" width="375" height="667"/>
```
Comment (khandpur) [inline Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib]: May want to add another scroll view container instead of modifying existing ones because the existing ones are already used in tests.

File: Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib [hunk 5]
```diff
@@ -19,66 +22,66 @@
...
             <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
             <subviews>
-                <textField opaque="NO" clipsSubviews="YES" contentMode="scaleToFill" misplaced="YES" contentHorizontalAlignment="left" contentVerticalAlignment="center" borderStyle="roundedRect" placeholder="Info / Insets {top,left,bottom,right}" clearsOnBeginEditing="YES" minimumFontSize="17" translatesAutoresizingMaskIntoConstraints="NO" id="P3e-SN-PxX">
-                    <rect key="frame" x="8" y="10" width="955" height="30"/>
+                <textField opaque="NO" clipsSubviews="YES" contentMode="scaleToFill" contentHorizontalAlignment="left" contentVerticalAlignment="center" borderStyle="roundedRect" placeholder="Info / Insets {top,left,bottom,right}" clearsOnBeginEditing="YES" minimumFontSize="17" translatesAutoresizingMaskIntoConstraints="NO" id="P3e-SN-PxX">
```
Comment (khandpur) [inline Tests/FunctionalTests/TestRig/Sources/FTRScrollViewController.xib]: Do you need all the other changes in this xib file to exhibit the bug?

**Comment by khandpur** [comment]
It seems like your commits may not be using the same email with which you signed the CLA. You may have to redo the commits using appropriate email.

**Comment by Ad1991** [comment]
Updated the pull request to remove unnecessary changes to the files and added new test cases instead of modifying the existing one.

**Comment by Ad1991** [comment]
I signed it!

**Comment by khandpur** [comment]
Adarsh, it could be that your git is configured to use different email. Can you check `git config --get user.email` returns the same email with which you signed the CLA?

File: Tests/FunctionalTests/Sources/FTRScrollViewTest.m [hunk 6]
```diff
@@ -28,6 +28,11 @@ - (void)setUp {
   [self openTestViewNamed:@"Scroll Views"];
 }
 
+
```
Comment (khandpur) [inline Tests/FunctionalTests/Sources/FTRScrollViewTest.m]: nit: remove excess new line

Comment (Ad1991) [reply Tests/FunctionalTests/Sources/FTRScrollViewTest.m]: Have a habit of keeping two newline between methods. Removed in next commit.

**Comment by Ad1991** [comment]
I have already changed the email to use the one I have signed the CLA with. Last commit was just to add the email information. Not sure why its failing. See attached images.

screen shot 2017-09-12 at 7 46 07 pm
screen shot 2017-09-12 at 7 44 29 pm
screen shot 2017-09-12 at 7 43 51 pm

**Comment by khandpur** [comment]
OK the Sep 8th's commit still shows it as "Adarsh Kumar Rai" not "Ad1991":
https://github.com/google/EarlGrey/pull/628/commits

you may need to squash all commits into one and repush just a single commit for this PR for it to update.

**Comment by Ad1991** [comment]
Squash and merge requires write permission to the repository as mentioned [here](https://help.github.com/articles/about-pull-request-merges/#squash-and-merge-your-pull-request-commits). Even rebase and merge seem to have requirement of write permission. Is there any chance you could squash the commits? If not, please decline this pull request and I will raise a separate pull request.

**Comment by khandpur** [comment]
ok, I'll close those one.
